### PR TITLE
MPDX-8306 Add phase grouping of statuses

### PR DESCRIPTION
**Pull request by caleballdrin**
## Description
https://jira.cru.org/secure/RapidBoard.jspa?rapidView=3&view=detail&selectedIssue=MPDX-8306&sprint=1268#
- Group statuses by phase in filters

## Checklist:

- [ ] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [ ] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [ ] I have requested a review from another person on the project

### CODE REVIEW

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:157]:
```suggestion
              )
```

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:144]: I think this could be clearer if we extracted `statusArray?.find((status) => status.id === option.value)` into a helper function that we called once for `a` and once for `b`.

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:133]: This is identical to the autocomplete besides the `groupBy` and `options` sorting, right? Ideally, we would extract it into a separate component that could be used in both places, and we pass in optional `groupBy` and `sortOptions` props.

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:130]: Do you mind adding a comment about the intention of this special case? Basically that we're grouping the status options by phase?

**Review by canac — APPROVED** [review]
I don't see any issues, so I'm going to approve. Sorting the options seems like something that should potentially be done on the server, so that we don't have to have as many special cases for certain filters. But we need a special case to group them, so it's not a big deal. And it's probably more work to add a `group` field on the server to the status options.

**Review by dr-bizz — COMMENTED** [review]
Since Caleb C has approved this, I'm good merging it once he approves it

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:116]: This looks like the same as line 123 and could be moved into a helper function.

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:114]:
```suggestion
  const groupByPhase = (option: string) =>
```

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [reply src/components/Shared/Filters/FilterListItemMultiselect.tsx:28]: Also, what do you think about naming this something like `StatusMultiselectFilter` because it is only used for statuses?

Comment (caleballdrin) [reply src/components/Shared/Filters/FilterListItemMultiselect.tsx:28]: @canac This component is used for both statuses and other Multiselect filters. It just renders differently based on the props given to it.

Comment (canac) [reply src/components/Shared/Filters/FilterListItemMultiselect.tsx]: My bad, you're right. You can ignore that suggestion about calling it a status filter.

File: src/components/Shared/Filters/FilterListItemMultiselect.tsx
Comment (canac) [inline src/components/Shared/Filters/FilterListItemMultiselect.tsx:78]:
```suggestion
              getStatusPhaseId(a.value)?.localeCompare(
                getStatusPhaseId(b.value) || '',
```

**Review by canac — APPROVED** [review]

**Pull request merged**
